addons/binding/org.openhab.binding.edimax/src/main/java/org/openhab/binding/edimax/internal/commands/AbstractCommand.java

```diff
@@ -0,0 +1,360 @@
+            ex.printStackTrace();
+        } catch (SAXException ex) {
+            ex.printStackTrace();
+        } catch (IOException ex) {
+            ex.printStackTrace();
+        }
+
+        return null;
+    }
+
+    /**
+     * Returns true if it is a SET command and false for GET.
+     *
+     * @return
+     */
+    protected boolean isSet() {
+        return (setValue != null);
+    }
+
+    /**
+     * Do the command.
+     *
+     * @param ci
+     * @return
+     * @throws IOException
+     */
+    public T executeCommand(ConnectionInformation ci) throws IOException {
+        String lastPart = "smartplug.cgi";
+        String response = executePost(ci.getUrl(), ci.getPort(), lastPart, getCommandString(), ci.getUsername(),
+                ci.getPassword());
+
+        return getResultValue(response);
+    }
+
+    private String executePost(String targetURL, int targetPort, String targetURlPost, String urlParameters,
+            String username, String password) throws IOException {
+        String complete = targetURL + ":" + targetPort + "/" + targetURlPost;
+
+        HttpURLConnection connection = null;
+        try {
+            // Create connection
+            URL url = new URL(complete);
+            connection = (HttpURLConnection) url.openConnection();
+            connection.setRequestMethod("POST");
+
+            connection.setRequestProperty("Connection", "Keep-Alive");
+            connection.setRequestProperty("Content-Type", "application/x-www-form-urlencoded");
+            connection.setRequestProperty("Content-Length", Integer.toString(urlParameters.getBytes().length));
+
+            String userpass = username + ":" + password;
+            String basicAuth = "Basic " + DatatypeConverter.printBase64Binary(userpass.getBytes());
+            connection.setRequestProperty("Authorization", basicAuth);
+
+            connection.setUseCaches(false);
+            connection.setDoOutput(true);
+
+            // Send request
+            DataOutputStream wr = new DataOutputStream(connection.getOutputStream());
+            wr.write(urlParameters.getBytes());
+            wr.close();
+
+            // Get Response
+            InputStream is = connection.getInputStream();
+            BufferedReader rd = new BufferedReader(new InputStreamReader(is));
+            StringBuilder response = new StringBuilder();
+            String line;
+            while ((line = rd.readLine()) != null) {
+                response.append(line);
+                response.append('\r');
+            }
+            rd.close();
+            return response.toString();
+        } finally {
+            if (connection != null) {
+                connection.disconnect();
+            }
+        }
+    }
+
+    /**
+     * Create initial XML tag.
+     *
+     * @param aName
+     * @return
+     */
+    protected String createStartTag(String aName) {
+        StringBuffer res = new StringBuffer();
```

Comment:
`StringBuilder` is faster than `StringBuffer` because it's not synchronized. So you should always use `StringBuilder`
 
And the thread safety which is delivered by `StringBuffer` is usually not enough because different append's can still be interleaved which still ruin your `StringBuffer`.

Just memorize to use the one which is the furthest in the Alphabet.